### PR TITLE
fix: continue improving Windows path escaping for commands

### PR DESCRIPTION
**Pull request by bwpge**
This PR addresses more edge cases with path escaping and vim commands, as noted in #1382.

After some testing, I found that only a small number of punctuation characters required the extra escape after  `vim.fn.fnameescape`. They are: `&`, `(`, `)`, `;`, `^`, and `` ` ``.

<details>
<summary>Details</summary>

Testing method used was running `:luafile %` with the following `test.lua` on a Windows machine:

```lua
-- copy/paste from neo-tree
local is_windows = vim.fn.has("win32") == 1 or vim.fn.has("win32unix") == 1

local escape_path_for_cmd = function(path)
  local escaped_path = vim.fn.fnameescape(path)
  if is_windows then
    for _, c in ipairs({ --[[ add characters as cases fail ]] }) do
      escaped_path = escaped_path:gsub("\\%" .. c, "\\%1")
    end
  end
  return escaped_path
end
-- end copy/paste

vim.fn.mkdir("tests", "p")

-- known allowed NTFS punctuation
local symbols = [[!#$%&'()+,-;=@[]^_`{}~ ]]

-- write file with a leading punctuation to a nested directory
for char in string.gmatch(symbols, ".") do
  -- double the punctuation to ensure it actually gets escaped correctly
  local f = string.format([[.\tests\%sfoo.txt]], char .. char)
  local p = escape_path_for_cmd(f)
  vim.cmd("silent w " .. p)
end

-- files outside of `tests` directory didn't get escaped correctly
for _, f in ipairs(vim.fn.glob("tests*.txt", false, true)) do
  vim.notify("Failed: " .. f, vim.log.levels.ERROR)
  os.remove(f)
end

-- these files were written to the correct path
for _, f in ipairs(vim.fn.glob("tests/*.txt", false, true)) do
  os.remove(f)
end
```

This testing logic could certainly be improved, I just quickly slapped this together.
</details>

The only reason I included the above is because it may be worth creating Windows path tests in the future and didn't want to lose this information.

I manually verified this fixes the issue in #1382, and tried to recreate some issues I've reported/worked on (such as #1352) and did not see any regressions. That being said, I would feel a bit better if someone could get a second pair of eyes on a Windows machine with this PR.

### CODE REVIEW

File: lua/neo-tree/utils/init.lua [hunk 1]
```diff
@@ -1018,11 +1018,19 @@ end
...
+    -- on windows, some punctuation preceeded by a `\` needs to have a second
+    -- `\` added to preserve the path separator. this is a naive replacement and
     -- definitely not bullet proof. if we start finding issues with opening files
-    -- or changing directories, look here first.
-    escaped_path = escaped_path:gsub("\\%p", "\\%1")
+    -- or changing directories, look here first. #1382 was the first regression
+    -- from the implementation that used lua's %p to match punctuation, which
+    -- did not quite work. the following characters were tested on windows to
+    -- be known to require an extra escape character.
+    for _, c in ipairs({ "&", "(", ")", ";", "^", "`" }) do
```
Comment (bwpge) [inline lua/neo-tree/utils/init.lua:1028]: Alternatively, we could just use a table with pre-escaped patterns like:

```lua
for _, c in ipairs({ "&", "%(", "%)", ";", "%^", "`" }) do
```

And then do the substitution with:

```lua
escaped_path = escaped_path:gsub("\\" .. c, "\\%1")
```

I'm just not intimately familiar with lua patterns (e.g., [magic characters](https://www.lua.org/pil/20.2.html)) so I used a what I thought might be the "safest" approach. If someone puts a `p` (or any other character class like `a`, `d`, etc.) in this table we'll have problems though... :)

**Comment by burgr033** [comment]
@bwpge asked me in #1382  to pull their fork and see if I find anything odd. I worked with their fork for half an hour and couldn't really find anything out of the ordinary.

I even checked with a few other characters.

```
_workspace
;workspace
(workspace
)workspace
&workspace
%workspace
`workspace
´workspace
^workspace
~workspace
```

Everything that I could check, works fine :)

File: lua/neo-tree/utils/init.lua (hunk 1 above)
Comment (cseickel) [reply lua/neo-tree/utils/init.lua:1028]: I think it reads clearer the way you have it now.

**Review by cseickel — APPROVED** [review]
Thanks for working on this!

**Comment by pysan3** [comment]
Thanks for working on this @bwpge !

Amazing work!

**Comment by bwpge** [comment]
Happy to help! And just a reminder to @burgr033 to change your config back, thanks for the assist :)

**Comment by burgr033** [comment]
thank you all <3

**Comment by pysan3** [comment]
@bwpge 

> @bwpge Pinging you again :)
> 
> We needed `" "` (check for space) at the end of the list in #1388. Right?
>
> From https://github.com/nvim-neo-tree/neo-tree.nvim/pull/1400#pullrequestreview-1939711304

**Pull request merged**